src/Widgets/BrightnessSlider.vala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2011-2016 elementary LLC. (https://elementary.io)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street - Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+public class Power.Widgets.BrightnessSlider : Gtk.Grid {
+    private const int STEP = 10;
+
+    private Gtk.Image image;
+    private Gtk.Scale scale;
+
+    public int val {
+        get { return (int) scale.get_value (); }
+        set { scale.set_value (value); }
+    }
+
+    // The integer returned is a relative change in the brightness value (ie.
+    // a +10 increase or a -10 decrease)
+    public signal void brightness_change (int change);
+
+    // The integer returned is absolute
```

Comment:
Integer is not strictly speaking 'returned' (it is a void function) - I presume you mean that the parameter must be positive?  Maybe could use a uint to ensure this?